src/moonlink_service/src/lib.rs

```diff
@@ -1,141 +1,66 @@
 mod error;
+pub mod rest_api;
+pub mod rpc_server;
 
-use arrow_ipc::writer::StreamWriter;
-pub use error::{Error, Result};
+pub use error::Result;
 use moonlink_backend::MoonlinkBackend;
 use moonlink_metadata_store::SqliteMetadataStore;
-use moonlink_rpc::{read, write, Request, Table};
-use std::collections::HashMap;
-use std::io::ErrorKind::{BrokenPipe, ConnectionReset, UnexpectedEof};
 use std::sync::Arc;
-use tokio::fs;
-use tokio::net::{UnixListener, UnixStream};
 use tokio::signal::unix::{signal, SignalKind};
+use tracing::{info, warn};
 
-pub async fn start(base_path: String) -> Result<()> {
+pub struct ServiceConfig {
+    pub base_path: String,
+    pub rest_api_port: Option<u16>,
+}
+
+pub async fn start_with_config(config: ServiceConfig) -> Result<()> {
```

Comment:
This is a breaking change, you should sync pg_mooncake.
https://github.com/Mooncake-Labs/pg_mooncake/blob/b2f68c5adf70b9da484e096899baab9a258f681c/src/bgworker.rs#L24